### PR TITLE
[RemoveDIs][NFC] Rename DPValue -> DbgVariableRecord

### PR DESCRIPTION
**Pull request by SLTozer**
This is the major rename patch that prior patches have built towards. The DPValue class is being renamed to DbgVariableRecord, which reflects the updated terminology for the "final" implementation of the RemoveDI feature. This is a pure string substitution + clang-format patch. The only manual component of this patch was determining where to perform these string substitutions: `DPValue` and `DPV` are almost exclusively used for DbgRecords, *except* for:

  - llvm/lib/target, where 'DP' is used to mean double-precision, and so appears as part of .td files and in variable names. NB: There is a single existing use of `DPValue` here that refers to debug info, which I've manually updated.
  - llvm/tools/gold, where 'LDPV' is used as a prefix for symbol visibility enums.

Outside of these places, I've applied several basic string substitutions, with the intent that they only affect DbgRecord-related identifiers; I've checked them as I went through to verify this, with reasonable confidence that there are no unintended changes that slipped through the cracks. The substitutions applied are all case-sensitive, and are applied in the order shown:

```
  DPValue -> DbgVariableRecord
  DPVal -> DbgVarRec
  DPV -> DVR
```

Following the previous rename patches, it should be the case that there are no instances of any of these strings that are meant to refer to the general case of DbgRecords, or anything other than the DPValue class. The idea behind this patch is therefore that pure string substitution is correct in all cases as long as these assumptions hold.

### CODE REVIEW

File: llvm/lib/IR/AsmWriter.cpp
Comment (OCHyams) [inline llvm/lib/IR/AsmWriter.cpp:2706]: Worth renaming `DPI`?

File: llvm/lib/Transforms/Utils/CloneFunction.cpp
Comment (OCHyams) [inline llvm/lib/Transforms/Utils/CloneFunction.cpp:279]: `RemapDbgVariableRecordRange` maybe should be `RemapDbgRecordRange`?

**Review by OCHyams — APPROVED** [review]
Some nits, but looks fine to merge once those are fixed.

**Comment by SLTozer** [comment]
Rebased onto current main following recent pushes, and addressed review comments,

**Review by OCHyams — APPROVED** [review]
LGTM

I haven't looked at the recent changes as closely as I did for the original patch, as they've been folded into the rebase (and the patch is 80 files), but I've given it a quick scan and it still SGTM.

**Comment by SLTozer** [comment]
Looks like one of the buildbots is failing after this landed: https://lab.llvm.org/buildbot/#/builders/193/builds/48663

I'm going to wait and see if it continues, because this patch _should_ be NFC, and I can't see an obvious way it'd lead to errors on that build for those tests specifically.

**Pull request merged**
